client/authority-discovery/src/lib.rs

```diff
@@ -64,7 +64,7 @@ use error::{Error, Result};
 use log::{debug, error, log_enabled, warn};
 use libp2p::Multiaddr;
 use sc_network::specialization::NetworkSpecialization;
-use sc_network::{DhtEvent, ExHashT};
+use sc_network::{DhtEvent, ExHashT, NetworkStateInfo};
```

Comment:
I feel like these things (trait + primitives) should rather be moved to `primitves`, but it's a separate discussion.

Reply:
You can check #4450